include/i_comm.hpp

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+namespace roboteq {
+
+    class i_comm {
+
+        public :
+
+        // i_comm();
+        i_comm() = delete;
```

Comment:
This is my bad. The constructor gets implicitly called when the derived constructor gets called, so it can't be deleted. What i was trying to achieve was that it should be impossible for the user to create an instance of this base class, and must instead create instances of the derived classes. this is already enforced with the presence of pure virtual methods.

Reply:
So just haave this be an empty constructor

Reply:
![image](https://user-images.githubusercontent.com/56097508/69999642-1e087500-1527-11ea-8f31-4757d11d75ed.png)

Reply:
it should be `i_comm(){}`